### PR TITLE
Improvement/wallet id check after shamir reconstruction/chi 4556

### PR DESCRIPTION
**Pull request by kartikCypherock**
This change enables wallet_id check after mnemonics is restored from the shamir shares

### CODE REVIEW

**Comment by rohanagarwal94** [comment]
Task linked: [CHI-4556 Improvement: Check wallet_id whenever mnemonics are generated from shamir shares](https://app.clickup.com/t/9002019994/CHI-4556)

**Review by ujjwal-cyph — APPROVED** [review]

**Review by ujjwal-cyph — APPROVED** [review]

File: src/controller_main.c [hunk 1]
```diff
@@ -438,6 +438,7 @@ static bool wallet_selector(uint8_t *data_array)
     for (; walletIndex < MAX_WALLETS_ALLOWED; walletIndex++) {
         if (memcmp(wallet_id, get_wallet_id(walletIndex), WALLET_ID_SIZE) == 0) {
             if (get_wallet_state(walletIndex) == VALID_WALLET) {
+                memcpy(wallet.wallet_id, wallet_id, WALLET_ID_SIZE);
```
Comment (ujjwal-cyph) [inline src/controller_main.c:441]:
```suggestion
                // further down the flow, this gets overwritten by the wallet_id fetched from cards
                memcpy(wallet.wallet_id, wallet_id, WALLET_ID_SIZE);
```

**Review by amanCypherock — CHANGES_REQUESTED** [review]
Change missing in restore wallet flow

Also, have we tested any positive and negative cases?
We can simulate negative cases with 
- Same name but different mnemonics wallet
- mnemonic generation form Incorrectly decrypted or non-decrypted shares

File: src/level_one/controller/controller_level_one.c [hunk 2]
```diff
@@ -148,6 +148,7 @@ void level_one_controller()
             }
             cy_exit_flow();
         }
+        memcpy(wallet.wallet_id, get_wallet_id(index), WALLET_ID_SIZE);
```
Comment (kartikCypherock) [inline src/level_one/controller/controller_level_one.c:151]:
```suggestion
        // further down the flow, this gets overwritten by the wallet_id fetched from cards
        memcpy(wallet.wallet_id, get_wallet_id(index), WALLET_ID_SIZE);
```

**Comment by kartikCypherock** [comment]
Moving this to draft state as there are side effects due to global `wallet` struct being re-populated by card data.

**Comment by ujjwal-cyph** [comment]
closing as it is stale now
